### ISSUE TITLE
WARNING: Project no longer maintained  - Do not use in vital solutions!

### ISSUE THREAD

**Issue by dxdbno** [react-pdf-viewer/react-pdf-viewer#1768]
Do not use or buy this product in vital solutions as it's no longer supported or maintained.

This project is no longer maintained by the owner - maybe due to  dead, sickness  or other - it's unknown.

**Comment by zoltansimari** [comment]
@dxdbno What makes you say this? While I see a lots of open issues with little to no response to them, there are many recent commits. As of writing the last one is 3 days old. @phuocng Could you please comment?

**Comment by dxdbno** [comment]
> @dxdbno What makes you say this? While I see a lots of open issues with little to no response to them, there are many recent commits. As of writing the last one is 3 days old. @phuocng Could you please comment?

A lot of people has tried to get in touch with @phuocng for a security fix, with no luck. He is the only maintainer on this project and the license prevents shared community development, @zoltansimari  hence not recommended for commercial or virtual use.

**Comment by melloware** [comment]
@dxdbno i could be wrong but it looks like @phuocng has make commits to main as recently as today?

**Comment by dxdbno** [comment]
@melloware Yes, he is making some changes from scss to css, while there are a lot of people asking for a security fix. 
IMO the wrong choice.
He has not answered any questions or issues in a long time. 
There has not been a release in over a year.
Good luck in getting a response from him, if you choose to use or buy this product.
It's your choice.

**Comment by janhoy** [comment]
Agree. A project that stops doing security releases for over a year, even for paid users, cannot be classified as maintained.

It is a classic pattern. One single maintainer, who burns out or for other reason drops the ball and goes radio silence. Even worse, numerous people have tried to reach out in  multiple channels, but zero response to customers, despite recent commits.

We cannot even know whether his github account has been hacked and that the recent commits is for a planned future malicious release.

I was hoping he'd be willing to pass the ball to the user community by open sourcing, #1765, but guess that was a bit too much to hope for :)

**Comment by felixhagspiel** [comment]
I have decided to contact gumroad directly (gumroad is where he sells the software: https://phuocng.gumroad.com/l/react-pdf-viewer?layout=profile ) and asked for a statement on how they handle software that is lo longer maintained. Will post here if I get an answer.

**Comment by felixhagspiel** [comment]
Unfortunately you can only report a creator in those cases: https://help.gumroad.com/article/286-how-do-i-report-a-gumroad-creator.html
But if you have purchased the product on gumroad, you can leave a review for the product :)

**Comment by creativar** [comment]
Curious what people are using as a replacement for this package now it is un-maintained? I have looked at many solutions and all are either: 

a. not as good 
b. a lot more work as more implementation work is involved (not as much out of the box functionality)
c. a lot more $$

**Comment by joshkay** [comment]
Also interested if anyone has a replacement for this lib.  Currently using it in one of my production apps, but would love to upgrade.

**Comment by jkgenser** [comment]
> Also interested if anyone has a replacement for this lib. Currently using it in one of my production apps, but would love to upgrade.

Hi @joshkay , I'm developing a 100% free and open source alternative. MIT license.

It doesn't have the full feature set of react-pdf-viewer yet but please take it for a spin and open an issue if there is a feature request you're looking for. 

https://github.com/jkgenser/react-pdf-headless

**Comment by andrewdoro** [comment]
Hey! We got tired of fighting with react-pdf-viewer at @unriddle-ai. After dealing with crashes, security issues, and general headaches, we decided to build something that just works.

Check out https://github.com/unriddle-ai/lector. It's basically a bunch of primitives for building pdf viewers:
- Virtualization, panning, and zooming built-in
- Open source MIT
- Headless architecture - composition pattern
- Experimental highlights, search, and thumbnails functionality

[See it in action](https://lector-weld.vercel.app/)

We're actively working on it, so a large part of the documentation is not ready. 

_It's currently used in production at Unriddle._

**Comment by melloware** [comment]
also just saw this one released and is still actively supported: https://www.react-pdf.dev/
